src/simulation/LoadGenerator.cpp

```diff
@@ -1743,4 +1813,32 @@ GeneratedLoadConfig::modeUploads() const
     return mode == LoadGenMode::SOROBAN_UPLOAD ||
            mode == LoadGenMode::MIXED_CLASSIC_SOROBAN;
 }
+
+std::pair<TxGenerator::TestAccountPtr, TransactionFrameBaseConstPtr>
+LoadGenerator::readTransactionFromFile(GeneratedLoadConfig const& cfg)
+{
+    ZoneScoped;
+
+    // Read the next transaction from the file
+    TransactionEnvelope txEnv;
+    if (!mPreloadedTransactionsFile.readOne(txEnv))
+    {
+        // End of file reached, start again from the beginning
+        throw std::runtime_error("LoadGenerator: End of file reached");
+    }
+
+    // Create a TransactionFrame from the envelope
+    auto txFrame = TransactionFrameBase::makeTransactionFromWire(
+        mApp.getNetworkID(), txEnv);
+
+    // Increment the sequence number of the source account
+    auto idx = mCurrPreloadedTransaction % cfg.nAccounts;
+    auto acc = mTxGenerator.getAccount(idx + cfg.offset);
+    releaseAssert(acc);
+    acc->setSequenceNumber(txFrame->getSeqNum());
+    ++mCurrPreloadedTransaction;
+
+    // Do not provide an account
+    return std::make_pair(nullptr, txFrame);
```

Comment:
Is there a specific reason we don't just return acc here? It looks like consumers of this pair assume that the `from` account is non-null, and we're just lucky we don't happen to hit those paths.

Reply:
The account is only needed for submission retries. In pre-generated mode, there's no retries, hence the account is not used. I didn't want to return the account here to avoid situations where we accidentally use incorrect cached info. Instead, we should fully rely on account information from the generated file (so if there's a bug somewhere, I'd prefer to crash).